lib/ansible/plugins/filter/core.py

```diff
@@ -493,6 +493,30 @@ def subelements(obj, subelements, skip_missing=False):
     return results
 
 
+def dig(data, *args):
+    '''Returns a value for a sequence of given keys/indexes into a structure,
+    such as an array or hash. The first encountered undefined value or key stops
+    the dig and None is returned. For instance {{ data | dig("key1", 2, "b" }}
+    is roughly equivalent to {{ data["key1"][2]["b"] }}. However a standard
+    index will return an error if any parent of the final key "b" is undefined,
+    the function will just return None in this case.
+
+    >>> data = {"key1": ["foo", "bar", {"a": 1234, "b": 5678}], "key2": "foo"}
+    >>> dig(data, "key1", 2, "b")
+    5678
+    >>> dig(data, "key3", 2, "b") is None
+    True
+
+    '''
+    result = data
+    for i in args:
+        try:
+            result = result[i]
+        except (TypeError, IndexError, KeyError) as e:
+            return None
```

Comment:
```suggestion
            return Undefined()
```

Reply:
And then add at the top:
```python
from jinja2.runtime import Undefined
```

Reply:
```suggestion
            return StrictUndefined(name=i)
```
You could also use StrictUndefined if you prefer to raise an exception when the Undefined value is being used. And use this:
```python
from jinja2.runtime import StrictUndefined
```

Reply:
BTW I would build up that name recursively, so that you can much better indicate what exactly failed. In my example above I only show the index/key that failed, but it's nicer if you show the complete path (e.g. complex_obj['orgs'][1]['users']['dag']) so you get:
```json
    "msg": "The task includes an option with an undefined variable. The error was: 'dag' is undefined"
```
instead of:
```json
    "msg": "The task includes an option with an undefined variable. The error was: obj['orgs'][1]['users']['dag'] is undefined"
```